File: api/model_api.py
import json
import requests

def handler(request):
    try:
        # Parse request JSON
        body = json.loads(request.body)
        text = body.get("text")
        if not text:
            return {
                "statusCode": 400,
                "body": json.dumps({"error": "No text provided"})
            }

        # Hugging Face Space API endpoint
        HF_URL = "https://huggingface.co/spaces/gabb11/sentiment-analyzer/api/predict/"

        payload = {"data": [text]}
        hf_response = requests.post(HF_URL, json=payload)
        hf_response.raise_for_status()

        # Return Hugging Face JSON directly
        return {
            "statusCode": 200,
            "body": hf_response.text,
            "headers": {"Content-Type": "application/json"}
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }

File: api/model_api.js
export default async function handler(req, res) {
  if (req.method !== "POST") return res.status(405).end();

  const { text } = req.body;

  if (!text) return res.status(400).json({ error: "No text provided" });

  try {
    const response = await fetch(
      "https://huggingface.co/spaces/gabb11/sentiment_analyzer/api/predict",
      {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ text })
      }
    );

    // Read raw prediction from Hugging Face Space
    const prediction = await response.json();

    // Return the same structure your frontend expects
    res.status(200).json(prediction);

  } catch (err) {
    console.error(err);
    res.status(500).json({ error: "Error connecting to Hugging Face Space API" });
  }
}
